src/hooks/useLoadDocuments.ts

```diff
@@ -1,9 +1,38 @@
 import useLoadFromEndpoint, { ReturnType } from './useLoadFromEndpoint'
-import { DocumentType, ENDPOINTS } from '../constants/endpoints'
+import { AlternativeWordType, DocumentType, ENDPOINTS } from '../constants/endpoints'
+import { ARTICLES } from '../constants/data'
+import { useEffect, useState } from 'react'
+
+export interface DocumentTypeFromServer {
+  id: number
+  word: string
+  article: number
+  document_image: Array<{ id: number; image: string }>
+  audio: string
+  alternatives: AlternativeWordType[]
+}
+
+const transformArticle = (documents: ReturnType<DocumentTypeFromServer[]>): ReturnType<DocumentType[]> => {
+  const documentsWithTransformedArticle: DocumentType[] = []
+  documents.data?.forEach(doc => {
+    const { article, ...rest } = doc
+    const formattedArticle = ARTICLES.find(article => {
+      return article.id === doc.article
+    })
+    if (!formattedArticle) {
+      throw new Error('Invalid Server Response')
+    }
+    const formattedDocument: DocumentType = { ...rest, article: formattedArticle }
+    documentsWithTransformedArticle.push(formattedDocument)
+  })
+  const { data, ...rest } = documents
+  return { ...rest, data: documentsWithTransformedArticle }
```

Comment:
Using `.map()` this becomes simpler :)

Reply:
yeah, strugggling a bit with this, because it changes the types of the object. I will have a closer look at this later (after deployment), as i do not have time and patience now for it.